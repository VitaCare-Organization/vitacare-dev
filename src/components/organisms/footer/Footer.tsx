import { Shield } from "lucide-react";

export default function Footer() {
  return (
    <footer className="flex flex-col md:flex-row justify-between items-center px-8 md:px-20 py-5 md:py-7 border-t border-[#E2E8F0] text-center md:text-left">
      <div className="flex items-center space-x-2">
        <Shield className="text-[#0096CC]" />
        <p className="font-bold text-xl text-[#020817]">VitaCare</p>
      </div>

      <p className="text-[#64748B] mt-2 md:mt-0">
        &copy; {new Date().getFullYear()} VitaCare. All rights reserved. Powered
        by Stellar blockchain.
      </p>

      <ul className="flex space-x-4 text-[#64748B] mt-2 md:mt-0 *:cursor-pointer">
        <li>Terms</li>
        <li>Privacy</li>
        <li>Contact</li>
      </ul>
    </footer>
  );
}
