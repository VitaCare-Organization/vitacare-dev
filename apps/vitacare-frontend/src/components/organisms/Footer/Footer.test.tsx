import { render, screen } from "@testing-library/react";
import Footer from "./Footer";

// Mock lucide-react icons
jest.mock("lucide-react", () => ({
  Shield: () => <div data-testid="shield-icon">Shield Icon</div>,
}));

describe("Footer Component", () => {
  it("renders the VitaCare brand correctly", () => {
    render(<Footer />);

    expect(screen.getByText("VitaCare")).toBeInTheDocument();
    expect(screen.getByTestId("shield-icon")).toBeInTheDocument();
  });

  it("displays the copyright notice with current year", () => {
    render(<Footer />);

    const currentYear = new Date().getFullYear();
    const copyrightText = `© ${currentYear} VitaCare. All rights reserved. Powered by Stellar blockchain.`;
    
    expect(screen.getByText(copyrightText)).toBeInTheDocument();
  });

  it("renders all footer links", () => {
    render(<Footer />);

    expect(screen.getByText("Terms")).toBeInTheDocument();
    expect(screen.getByText("Privacy")).toBeInTheDocument();
    expect(screen.getByText("Contact")).toBeInTheDocument();
  });

  it("has correct footer structure and styling", () => {
    render(<Footer />);

    const footer = screen.getByRole("contentinfo");
    expect(footer).toHaveClass("flex flex-col md:flex-row justify-between items-center px-8 md:px-20 py-5 md:py-7 border-t border-[#E2E8F0] text-center md:text-left");
  });

  it("brand section has correct styling", () => {
    render(<Footer />);

    const brandSection = screen.getByText("VitaCare").closest("div");
    expect(brandSection).toHaveClass("flex items-center space-x-2");
    
    const brandText = screen.getByText("VitaCare");
    expect(brandText).toHaveClass("font-bold text-xl text-[#020817]");
  });

  it("copyright text has correct styling", () => {
    render(<Footer />);

    const currentYear = new Date().getFullYear();
    const copyrightElement = screen.getByText(`© ${currentYear} VitaCare. All rights reserved. Powered by Stellar blockchain.`);
    
    expect(copyrightElement).toHaveClass("text-[#64748B] mt-2 md:mt-0");
  });

  it("footer links have correct styling and cursor pointer", () => {
    render(<Footer />);

    const linksList = screen.getByText("Terms").closest("ul");
    expect(linksList).toHaveClass("flex space-x-4 text-[#64748B] mt-2 md:mt-0 *:cursor-pointer");
  });

  it("links are clickable elements", () => {
    render(<Footer />);

    const termsLink = screen.getByText("Terms");
    const privacyLink = screen.getByText("Privacy");
    const contactLink = screen.getByText("Contact");

    // Check they are list items (clickable)
    expect(termsLink.tagName).toBe("LI");
    expect(privacyLink.tagName).toBe("LI");
    expect(contactLink.tagName).toBe("LI");
  });

  it("has proper semantic structure", () => {
    render(<Footer />);

    // Should be a footer element
    const footer = screen.getByRole("contentinfo");
    expect(footer.tagName).toBe("FOOTER");
  });

  it("displays current year dynamically", () => {
    render(<Footer />);

    const currentYear = new Date().getFullYear();
    expect(screen.getByText(new RegExp(`© ${currentYear}`))).toBeInTheDocument();
  });
});
