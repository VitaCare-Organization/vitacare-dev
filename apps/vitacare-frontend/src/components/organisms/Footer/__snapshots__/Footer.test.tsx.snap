// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component matches snapshot 1`] = `
<footer
  class="flex flex-col md:flex-row justify-between items-center px-8 md:px-20 py-5 md:py-7 border-t border-[#E2E8F0] text-center md:text-left"
>
  <div
    class="flex items-center space-x-2"
  >
    <div
      data-testid="shield-icon"
    >
      Shield Icon
    </div>
    <p
      class="font-bold text-xl text-[#020817]"
    >
      VitaCare
    </p>
  </div>
  <p
    class="text-[#64748B] mt-2 md:mt-0"
  >
    Â© 
    2025
     VitaCare. All rights reserved. Powered by Stellar blockchain.
  </p>
  <ul
    class="flex space-x-4 text-[#64748B] mt-2 md:mt-0 *:cursor-pointer"
  >
    <li>
      Terms
    </li>
    <li>
      Privacy
    </li>
    <li>
      Contact
    </li>
  </ul>
</footer>
`;
